gui/builtinContextMenus/metaSwap.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+from gui.contextMenu import ContextMenu
+from gui.itemStats import ItemStatsDialog
+import gui.mainFrame
+import service
+import wx
+import gui.globalEvents as GE
+
+class MetaSwap(ContextMenu):
+    def __init__(self):
+        self.mainFrame = gui.mainFrame.MainFrame.getInstance()
+
+    def display(self, srcContext, selection):
+
+        if self.mainFrame.getActiveFit() is None or srcContext not in ("fittingModule",):
+            return False
+
+        # Check if list of variations is same for all of selection
+        # If not - don't show the menu
+        mkt = service.Market.getInstance()
+        self.variations = None
+        for i in selection:
+            variations = mkt.getVariationsByItems([i.item])
+            if self.variations is None:
+                self.variations = variations
+            else:
+                if variations != self.variations:
+                    return False
+
+        self.selection = selection
+
+        return True
+
+    def getText(self, itmContext, selection):
+        return "Variations"
+
+    def getSubMenu(self, context, selection, rootMenu, i, pitem):
+        self.moduleLookup = {}
+
+        def get_metalevel(x):
+            return x.attributes["metaLevel"].value
+
+        def get_metagroup(x):
+            return x.metaGroup.ID if x.metaGroup is not None else 0
+
+        m = wx.Menu()
+
+        # Sort items by metalevel, and group within that metalevel
+        items = list(self.variations)
+        items.sort(key=get_metalevel)
+        items.sort(key=get_metagroup)
+
+        group = None
+        for item in items:
+            # Apparently no metaGroup for the Tech I variant:
+            if item.metaGroup is None:
+                thisgroup = "Tech I"
+            else:
+                print item.metaGroup.ID
```

Comment:
Seems to work fine, I like it a lot. The context menu can really increase the speed you can work with pyfa.

Btw you still have a "print" in gui/builtinContextMenus/metaSwap.py:59